### DISCUSSION TITLE
ch4-rpc/ch4-03-netrpc-hack

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#620] [General]
# ch4-rpc/ch4-03-netrpc-hack

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch4-rpc/ch4-03-netrpc-hack.html

**Comment by Wu-Fisher** [comment]
作者你好，kvs设计的watch是不是怪怪的。过滤器检测到了以后如果后续保留着，拿后续如果发生十次的再次修改调用，ch缓冲区是不是就会满。还有clientwork的时候，可能set在watch之前调用，首次可能观察不到。emmm这是我目前阅读感到不是很符合逻辑的地方，如果有我理解不对的地方麻烦也老师纠正一下🙏

**Reply by ceyewan** [reply]
缓冲区可能会满，这只是教学代码，超时了结束程序就行。“可能set在watch之前调用，首次可能观察不到”，是这样的，我复现的时候就是观察不到，睡一秒之后再 set 就行了。

**Comment by changkaiyan** [comment]
您好，我将尽快回复您的邮件。如果您的邮件无需回复，则收到本回复表明我已收到您的邮件。请勿回复，谢谢合作。This is an automatic reply, confirming that your e-mail was received.Thank you.
